src/common/snippets/src/lowered/pass/propagate_subtensors.cpp

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/propagate_subtensors.hpp"
+
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/utils.hpp"
+#include "snippets/itt.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+namespace {
+void propagate_updated_subtensor_through_loop(const LinearIR& linear_ir,
+                                              const LinearIR::LoopManager::LoopInfoPtr& loop_info,
+                                              LinearIR::container::const_iterator begin,
+                                              LinearIR::container::const_iterator end,
+                                              const size_t new_dim_value) {
+    std::map<lowered::PortDescriptorPtr, snippets::VectorDims> original_shapes;
+    static constexpr size_t existing_subtensor_value = SIZE_MAX;
```

Comment:
We have quite a few of SIZE_MAX hardcoded constants. Is this somehow related to `FULL_DIM`?

Reply:
In this case, this constant is used as some default `new_dim_value` which is used in inner loops. It was agreed offline that it's better to add a bool parameter which will be more descriptive than the current version.